rust/tool/src/install.rs

```diff
@@ -316,6 +316,7 @@ impl Installer {
         )
         .context("Failed to assemble lanzaboote image.")?;
 
+        self.install_systemd_boot(&generation.spec.bootspec.system, &esp_gen_paths)?;
```

Comment:
Calling this function here means that for every generation, the systemd-boot binary on the ESP AND in the Nix Store is at least opened and checked. This seems pretty wasteful :D

Also, if we allow/treat the cursed case of generations with different architectures/systems, we cannot write the path to the systemd-boot binary into the efivars anymore.

All in all I'm more and more convinced that we should not allow/treat generations with different architectures. We should add system as a parameter to `lzbt`:

Reply:
Good point; let's hard fail on multiple systems in generations and write the unique systemd boot once based on the unique value of system, ok for you?

Reply:
(actually we don't care that much about efivars because we have EFI fallback dirs technically)

Reply:
> Good point; let's hard fail on multiple systems in generations and write the unique systemd boot once based on the unique value of system, ok for you?

So you still want to read the system from the bootspec?

Reply:
Yes because they are the real source of truth IMHO, if they have anomalous data, it's a problem we should report and we should not act based on external overrides from command line (yet?) IMHO.

Reply:
I wouldn't do it because (a) this would complicate the code quite a bit, (b) we already read systemd from the command line  so we might as well do it for `system` (this would be effectively what systemd-boot-builder.py does but systemd-boot-builder.py relies on systemd being built for the right target), and (c) this would provide little benefit to the user.

Reply:
Fair points. Let's go with your suggestion for now.

Reply:
Done